### PR TITLE
Fix:FM-210 Optimized caching of language assets and audio files in se…

### PR DESCRIPTION
**Pull request by ashwinnair-chimple**
…rvice worker with deduplication and progressive loading

# Changes
- Implementing optimized caching of language assets and audio files in service worker with duplication and progressive loading

# How to test
- Open the application the caching process during the initialization will check for duplication and add only unique files to cache storage.

Ref: [FM-210](https://curiouslearning.atlassian.net/browse/FM-210)

[FM-210]: https://curiouslearning.atlassian.net/browse/FM-210?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: sw-src.js [hunk 1]
```diff
@@ -55,99 +45,57 @@ self.registration.addEventListener("updatefound", function (e) {
...
 }
 
 async function getALLAudioUrls(cacheName, language) {
-  // await cacheCommonAssets(language);
-  let audioList = [];
-  audioList.push("/lang/" + language + "/ftm_" + language + ".json");
+  let audioList = new Set(); // Use Set to filter duplicates
+  audioList.add("/lang/" + language + "/ftm_" + language + ".json");
+
   fetch("./lang/" + language + "/ftm_" + language + ".json", {
```
Comment (woodsjohn-codev) [inline sw-src.js:72]: this could improve into `./lang/${language}/ftm_${language}.json`

**Review by woodsjohn-codev — COMMENTED** [review]
Looks good overall just one tiny comment

File: sw-src.js [hunk 2]
```diff
@@ -55,99 +45,57 @@ self.registration.addEventListener("updatefound", function (e) {
...
-        //  await cacheAudiosFiles(
-        //     element.prompt.PromptAudio,
-        //     workbox.core.cacheNames.precache + language,
-        //     data.Levels.length
-        //   );
-        });
+      await cacheFeedBackAudio(data.FeedbackAudios, language);
+
+      for (const level of data.Levels) {
+        for (const puzzle of level.Puzzles) {
```
Comment (bernhardccodev) [inline sw-src.js:82]: Question regarding the for loop, for the data.levels is this the 150 game levels that has 5 game puzzles?

File: sw-src.js (hunk 1 above)
Comment (ashwinnair-chimple) [reply sw-src.js:72]: sure will consider this

**Pull request merged**